### PR TITLE
[IMP] readme: add the Odoo workflow

### PR DESCRIPTION
**Pull request by d-fence**

### CODE REVIEW

File: runbot_merge/README.rst [hunk 1]
```diff
@@ -1,6 +1,42 @@
 Merge Bot
 =========
 
+Odoo workflow
+-------------
+
+The sticky branches are protected on the github odoo project to restrict
+push for the Merge Bot only.
+
+The MB only works with PR's using the github API.
```
Comment (xmo-odoo) [inline runbot_merge/README.rst]: PRs. Also for readability maybe add `(MB)` after the "Merge Bot" in the paragraph above so it's clear what MB means afterwards.

Alternatively, we could use a [substitution](http://docutils.sourceforge.net/docs/user/rst/quickref.html#substitution-references-and-definitions) to write `|MB|` in the text, but have `Merge Bot` or `the merge bot` in the output everywhere.

File: runbot_merge/README.rst [hunk 2]
```diff
@@ -1,6 +1,42 @@
...
+
+
+1. When a PR is created the github webhook will signal it to the MB. The
```
Comment (xmo-odoo) [inline runbot_merge/README.rst]: "the github webhook will signal it to" -> "github notifies".

Also a note (possibly after the initial list): if there's activity on a PR the mergebot hasn't seen yet, it'll build a job to fetch it so e.g. a comment on a PR will bring it to the mergebot's attention.

File: runbot_merge/README.rst [hunk 3]
```diff
@@ -1,6 +1,42 @@
...
+   MB will label the PR as 'seen 🙂' on github.
+
+2. Once the PR github statuses are green (Runbot build is green and CLA
+   is signed if needed), the MB will label the PR as 'CI 🤖'.
+
+3. When a reviewer, known by the MB, will approve the PR, the MB will
+   label the PR as 'r+ 👌'.
```
Comment (xmo-odoo) [inline runbot_merge/README.rst]: use present tense e.g. "reviewer approves", "MB labels".

File: runbot_merge/README.rst [hunk 4]
```diff
@@ -1,6 +1,42 @@
...
+4. At this moment, MB will try to merge the PR and will label the PR
+   with 'merging 👷'.
+
+5. If the merge is successfull, LB will label it 'merged 🎉', remove the
+   label 'merging 👷' and close the PR. A message from MB will give a
+   link to the merge commit.
+
+If an error occurs during the process, MB will label the PR with 'error
+🙅' and add a message in the conversion stating what kind of error.  For
+example 'Unable to stage PR (merge conflict)'.
```
Comment (xmo-odoo) [inline runbot_merge/README.rst]: Technically it's only if an error occurs during step 4, I think.

File: runbot_merge/README.rst [hunk 5]
```diff
@@ -1,6 +1,42 @@
...
+push for the Merge Bot only.
+
+The MB only works with PR's using the github API.
+
+
+1. When a PR is created the github webhook will signal it to the MB. The
+   MB will label the PR as 'seen 🙂' on github.
+
+2. Once the PR github statuses are green (Runbot build is green and CLA
+   is signed if needed), the MB will label the PR as 'CI 🤖'.
```
Comment (xmo-odoo) [inline runbot_merge/README.rst]: Parenthetical should probably be a footnote instead, and it's only an example: we can customise which statuses are expected.

**Comment by d-fence** [comment]
@xmo-odoo changes done

File: runbot_merge/README.rst [hunk 6]
```diff
@@ -1,6 +1,52 @@
...
+   MB's attention. e.g a comment on a PR.
+
+.. [#] At this moment the statuses are: Runbot build is green and CLA is
+   signed if needed.  The expected statuses may change in the future.
+
+.. [#] If a PR contains only one commit, the PR is rebased and the
+   commit is fast forwarded. With more than one commit, the PR is
+   rebased and the commits are merged with a merge commit. When one
+   wants to avoid the rebase, 'rebase-' command should be used.
+
```
Comment (d-fence) [inline runbot_merge/README.rst]: @xmo-odoo I added this paragraph to explain FF/merge

**Pull request merged**